### DISCUSSION TITLE
Where can I set the authorization for the administration/configuration endpoint so that users with, lets say, admin claim only can access it?

### DISCUSSION THREAD

**Discussion by aamuz** [ThreeMammals/Ocelot#1814] [Q&A]
### [Question] Authorization with external IdentityServer

When Ocelot is used with and external IdentityServer, everything is awesome and works like expected. However, what i haven't figured out from the documentation is where can I set the authorization for the administration/configuration endpoint so that users with, lets say, admin claim only can access it.

Right now all identities that passes the authentication has access to that endpoint, means they can change the reroute config. I must have missing something simple.

**Comment by TomPallister** [comment]
@aamuz Thanks for your interest in the project! Unfortunately you haven't missed anything simple! There is no claims based authorisation on the administration API at the moment. I take a lean approach to development so until someone asks for a feature I normally won't add it! 

So I agree this is required and will add it to the backlog.

I think something like [this](https://docs.microsoft.com/en-us/aspnet/core/security/authorization/policies?view=aspnetcore-2.1) should allow us to implement something dynamic so users to implement their own authorisation policies for the administration endpoint. 

This is on the backlog but i have lots to do at the moment and am struggling just to keep up with questions and bugs with my spare time but I will get round to it eventually. If you have time to give it a go yourself I welcome PRs and am happy to support with design ideas etc!

**Comment by aamuz** [comment]
@TomPallister understood.

I will look into the code and will definitely try to see if I can add this enhancement and send you a PR. 

I, however, would definitely need some ideas from you.

**Comment by TomPallister** [comment]
@aamuz thats OK let me know when / if you need support!

**Comment by tmkhan** [comment]
@TomPallister @aamuz 
I am also struggling with the same issue. Please help me if any update on this. Thanks in advance.

**Comment by raman-m** [comment]
@aamuz 
Are you still with Ocelot?
